src/layout/HeaderSearch.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { ChangeEvent, useState } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { useDebounceFn } from 'ahooks';
+import { TextInput } from 'carbon-components-react';
+
+import './HeaderSearch.scss';
+
+function HeaderSearch({ }: { }) {
+  const location = useLocation();
+  const { pathname } = location;
+  const pathKey = '/search/'
+  const initKeyword = pathname.includes(pathKey) ? pathname.substr(pathname.indexOf(pathKey) + pathKey.length) : '';
+
+  const [keyword, setKeyword] = useState(initKeyword);
+  const history = useHistory();
+
+  const handleSearchThrottled = useDebounceFn(
+    () => {
+      if (!keyword || keyword === '') {
+        history.push('/');
+      } else {
+        history.push(`${pathKey}${keyword}`);
+      }
+    },
+    { wait: 1000 },
+  );
+
+  const onChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setKeyword(value);
+    handleSearchThrottled.run();
+  }
+
+  return (
+    <div className="HeaderSearch">
+      <TextInput
+        id="search-input"
+        invalidText="A valid value is required"
+        labelText=""
+        placeholder="Please input search keyword"
+        size="sm"
+        onChange={onChange}
+        defaultValue={initKeyword}
```

Comment:
how about handle search after Enter or clicking search button or something similar? Then you don't need this debounceFn any more.